src/adapters/amqp.js

```diff
@@ -280,35 +280,69 @@ class AmqpAdapter extends BaseAdapter {
 			chan.id
 		);
 
+		const exchangeOptions = _.defaultsDeep(
+			{},
+			chan.amqp ? chan.amqp.exchangeOptions : {},
+			this.opts.amqp.exchangeOptions
+		);
+
+		const queueOptions = _.defaultsDeep(
+			{},
+			chan.amqp ? chan.amqp.queueOptions : {},
+			this.opts.amqp.queueOptions
+		);
+
+		chan.deadLettering = _.defaultsDeep({}, chan.deadLettering, this.opts.deadLettering);
+
+		const queueName = `${chan.group}.${chan.name}`;
+
 		try {
 			if (chan.maxRetries == null) chan.maxRetries = this.opts.maxRetries;
-			chan.deadLettering = _.defaultsDeep({}, chan.deadLettering, this.opts.deadLettering);
+
 			if (chan.deadLettering.enabled) {
 				chan.deadLettering.queueName = this.addPrefixTopic(chan.deadLettering.queueName);
 				chan.deadLettering.exchangeName = this.addPrefixTopic(
 					chan.deadLettering.exchangeName
+						? chan.deadLettering.exchangeName
+						: "DEAD_LETTER"
+				);
+
+				this.logger.debug(`Asserting exchange ${chan.deadLettering.exchangeName}`);
+				this.assertedExchanges.add(chan.deadLettering.exchangeName);
+				await this.channel.assertExchange(
```

Comment:
Please change to 
```js
await this.channel.assertExchange(
					chan.deadLettering.exchangeName,
					"fanout",
					_.defaultsDeep(
						{},
						chan.deadLettering ? chan.deadLettering.exchangeOptions : {},
						this.opts.exchangeOptions
					)
				);
```

This way we can provide custom opts via `chan.deadLettering.exchangeOptions`

Reply:
OK.